### PR TITLE
Refs #27086 -- Corrections to unit testing contributor docs. 

### PR DESCRIPTION
**Pull request by carltongibson**
In the [unit tests contributor docs](https://docs.djangoproject.com/en/2.2/internals/contributing/writing-code/unit-tests/), the _Tips_ section is over-indented under the _Troubleshooting_ section. It should be a sibling. (I think.) 

This just outdents it a notch. 

#### Before:

before

#### After:

after

### CODE REVIEW

**Comment by carltongibson** [comment]
Hey @felixxm — can you take another look please? Added two commits: 

* My other TODO from DjangoCon sprints. (_Upgrade Python!_)
* Cherry picked @khink's commit from  #11203

Should wrap up those 🤞

Ta!

File: docs/internals/contributing/writing-code/unit-tests.txt [hunk 1]
```diff
@@ -380,6 +380,20 @@ Ensure you have the latest point release of a :ref:`supported Python version
...
+On **macOS** (High Sierra and newer versions), you might see this message
+logged, after which the tests hang::
+
+    objc[42074]: +[__NSPlaceholderDate initialize] may have been in progress in
+    another thread when fork() was called.
+
+To avoid this set a ``OBJC_DISABLE_INITIALIZE_FORK_SAFETY`` environment
+variable, for example::
+
+    OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ./runtests.py
```
Comment (felixxm) [inline docs/internals/contributing/writing-code/unit-tests.txt]:
```
    $ OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ./runtests.py 
```

File: docs/internals/contributing/writing-code/unit-tests.txt [hunk 2]
```diff
@@ -380,6 +380,20 @@ Ensure you have the latest point release of a :ref:`supported Python version
...
+
+Or by adding `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` to your
+`~/.profile`, or equivalent shell startup script.
```
Comment (felixxm) [inline docs/internals/contributing/writing-code/unit-tests.txt]: Please use double backticks.

Comment (felixxm) [reply docs/internals/contributing/writing-code/unit-tests.txt]: Maybe:
```
Or by adding ``export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES`` to your shell's
startup file (e.g. ``~/.profile``).
```

**Comment by carltongibson** [comment]
Super. Thanks for the review @felixxm!

**Pull request merged**